src/aliceVision/feature/apriltag/ImageDescriber_APRILTAG.cpp

```diff
@@ -0,0 +1,117 @@
+#include <aliceVision/system/Logger.hpp>
+
+#include <apriltag/apriltag.h>
+#include <apriltag/tag16h5.h>
+#include <opencv2/core/core.hpp>
+
+namespace aliceVision {
+namespace feature {
+
+ImageDescriber_APRILTAG::AprilTagParameters::AprilTagParameters()
+{
+}
+
+void ImageDescriber_APRILTAG::AprilTagParameters::setPreset(EImageDescriberPreset preset)
+{
+}
+
+ImageDescriber_APRILTAG::ImageDescriber_APRILTAG()
+  : ImageDescriber()
+  {}
+
+void ImageDescriber_APRILTAG::allocate(std::unique_ptr<Regions> &regions) const
+{
+  regions.reset( new APRILTAG_Regions );
+}
+
+void ImageDescriber_APRILTAG::setConfigurationPreset(ConfigurationPreset preset)
+{
+  _params.setPreset(preset.descPreset);
+}
+
+EImageDescriberType ImageDescriber_APRILTAG::getDescriberType() const
+{
+  return EImageDescriberType::APRILTAG16H5;
+}
+
+bool ImageDescriber_APRILTAG::describe(const image::Image<unsigned char>& image,
+    std::unique_ptr<Regions> &regions,
+    const image::Image<unsigned char> * mask)
+{
+  allocate(regions);
+  ALICEVISION_LOG_DEBUG(" regions.get() = " << regions.get());
+  APRILTAG_Regions * regionsCasted = dynamic_cast<APRILTAG_Regions*>(regions.get());
+  ALICEVISION_LOG_DEBUG(" regionsCasted = " << regionsCasted);
+
+  apriltag_family_t *tf = tag16h5_create();
+  apriltag_detector_t *td = apriltag_detector_create();
+  apriltag_detector_add_family(td, tf);
+  td->quad_decimate = 2.0;
+  td->quad_sigma = 0.0; // blur
+  td->nthreads = 1;
+  td->debug = 0;
+  td->refine_edges = 1;
+
+  const cv::Mat graySrc(cv::Size(image.Width(), image.Height()), CV_8UC1, (unsigned char *) image.data(), cv::Mat::AUTO_STEP);
+  // Make an image_u8_t header for the Mat data
+  image_u8_t im = { .width = graySrc.cols,
+      .height = graySrc.rows,
+      .stride = graySrc.cols,
+      .buf = graySrc.data
+  };
+  zarray_t *detections = apriltag_detector_detect(td, &im);
+  // Draw detection outlines
+  for (int i = 0; i < zarray_size(detections); i++) {
+    apriltag_detection_t *det;
+    zarray_get(detections, i, &det);
+    if (det->hamming == 0) {
+      // compute center point as the intersection of the two diagonals
+      // note: AprilTag corner points are listed counter-clockwise:
+      cv::Point2d tl(det->p[0][0], det->p[0][1]);
+      cv::Point2d bl(det->p[1][0], det->p[1][1]);
+      cv::Point2d br(det->p[2][0], det->p[2][1]);
+      cv::Point2d tr(det->p[3][0], det->p[3][1]);
+      double denominator = ((tl.x-br.x) * (bl.y-tr.y) - (tl.y-br.y) * (bl.x-tr.x));
+      cv::Point2d center(
+        ((tl.x*br.y - tl.y*br.x) * (bl.x-tr.x) - (tl.x-br.x) * (bl.x*tr.y - bl.y*tr.x)) / denominator,
+        ((tl.x*br.y - tl.y*br.x) * (bl.y-tr.y) - (tl.y-br.y) * (bl.x*tr.y - bl.y*tr.x)) / denominator
+      );
+      cv::Point2d points[5] = { center, tl, bl, br, tr };
+      std::size_t indices[5] = { det->id, 30 + det->id, 60 + det->id, 90 + det->id, 120 + det->id};
+      // compute scale from max side length and diagonals (divided by sqare root of 2):
```

Comment:
same here, cv:Point2d can be replaced by `Vec2`, less handy to manage (no .x and .y) but it eases things for the dependencies